src/server/chromium/crBrowser.ts

```diff
@@ -503,10 +504,18 @@ export class CRBrowserContext extends BrowserContext {
     return Array.from(this._browser._serviceWorkers.values()).filter(serviceWorker => serviceWorker._browserContext === this);
   }
 
-  async newCDPSession(page: Page): Promise<CRSession> {
-    if (!(page instanceof Page))
-      throw new Error('page: expected Page');
-    const targetId = (page._delegate as CRPage)._targetId;
+  async newCDPSession(target: Page | Frame): Promise<CRSession> {
+    let targetId: string | null = null;
+    if (target instanceof Page) {
+      targetId = (target._delegate as CRPage)._targetId;
+    } else if (target instanceof Frame) {
+      const session = (target._page._delegate as CRPage)._sessions.get(target._id);
+      if (!session) throw new Error('No session for Target ID. This frame might be part of another target');
+      targetId = session._targetId;
+    } else {
+      throw new Error('target: expected Page or Frame');
+    }
+
```

Comment:
All this could be changed to:
```
const targetId = (target instanceof Page ? target.mainFrame() : target)._id
```
if we don't mind getting `browserContext.newCDPSession: Protocol error (Target.attachToTarget): No target with given id found` for frames that are part of another target. (Since the original code didn't do this, I went the more explicit _targetId finding route).